daml-lf/tests/MultiKeys.daml

```diff
@@ -0,0 +1,232 @@
+         None <- lookupByKey @Keyed p
+         pure ()
+
+    -- should be accepted in uck
+    nonconsuming choice GlobalArchiveOverwritesKnownGlobal1 : ()
+      with
+        cid1 : ContractId Keyed
+        cid2 : ContractId Keyed
+      controller p
+      do cid1 =/= cid2
+         c1 <- fetch cid1
+         key c1 === p
+         c2 <- fetch cid2
+         key c2 === p
+         (keyCid, _) <- fetchByKey @Keyed p
+         assert (keyCid `elem` [cid1, cid2])
+         -- A consuming choice is
+         -- implicitly a by-key operation.
+         archive (fromSome (find (/= keyCid) [cid1, cid2]))
+         None <- lookupByKey @Keyed p
+         pure ()
+
+    -- should be accepted in uck
+    nonconsuming choice GlobalArchiveOverwritesKnownGlobal2 : ()
+      with
+        cid1 : ContractId Keyed
+        cid2 : ContractId Keyed
+      controller p
+      do cid1 =/= cid2
+         c1 <- fetch cid1
+         key c1 === p
+         c2 <- fetch cid2
+         key c2 === p
+         (keyCid, _) <- fetchByKey @Keyed p
+         assert (keyCid `elem` [cid1, cid2])
+         archive keyCid
+         None <- lookupByKey @Keyed p
+         pure ()
+
+    -- should be rejected in uck
+    nonconsuming choice RollbackCreateNonRollbackFetchByKey : ()
+      controller p
+      do try do
+           create (Keyed p)
+           throw E
+         catch E -> pure ()
+         _ <- fetchByKey @Keyed p
+         pure ()
+
+    -- should be rejected in uck
+    nonconsuming choice RollbackFetchByKeyNonRollbackCreate : ()
+      controller p
+      do try do
+           _ <- fetchByKey @Keyed p
+           throw E
+         catch E -> pure ()
+         cid <- create (Keyed p)
+         (cid', _) <- fetchByKey @Keyed p
+         cid === cid'
+
+    -- should be accepted in uck
+    nonconsuming choice RollbackFetchNonRollbackCreate : ()
+      with
+        cid : ContractId Keyed
+      controller p
+      do try do
+           c <- fetch cid
+           key c === p
+           throw E
+         catch E -> pure ()
+         cid <- create (Keyed p)
+         (cid', _) <- fetchByKey @Keyed p
+         cid === cid'
+
+    -- should be rejected in uck
+    nonconsuming choice RollbackGlobalArchiveNonRollbackCreate : ()
+      with
+        cid : ContractId Keyed
+      controller p
+      do try do
+           c <- fetch cid
+           key c === p
+           -- Again, consuming choices are implicitly by-key.
+           archive cid
+           throw E
+         catch E -> pure ()
+         cid <- create (Keyed p)
+         (cid', _) <- fetchByKey @Keyed p
+         cid === cid'
+
+    -- should be rejected in uck
+    nonconsuming choice RollbackCreateNonRollbackGlobalArchive : ()
+      with
+        cid : ContractId Keyed
+      controller p
+      do try do
+           create (Keyed p)
+           throw E
+         catch E -> pure ()
+         c <- fetch cid
+         key c === p
+         archive cid
+         None <- lookupByKey @Keyed p
```

Comment:
I disagree here. Like for `GlobalArchiveOverwritesUnknownGlobal`, Daml engine has not yet seen a read of the key from the underlying ledger, so the outcome depends on the response to the `ResultNeedKey`:
* If it's `Some cid`, then `None` is the `lookupByKey` result.
* If it's `Some otherCid`, then `Some otherCid` is the result.

Reply:
As discussed, we’re postponing multikey semantics so resolving this comment for this PR.